### PR TITLE
[zh] Sync egress SNI routing blog into Chinese

### PR DESCRIPTION
**Pull request by wilsonwu**
## Description

Sync egress SNI routing blog into Chinese

## Reviewers

- [ ] Ambient
- [x] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [x] Localization/Translation

### CODE REVIEW

**Review by windsonsea — COMMENTED** [review]
这个插图的文字真多，能转成可编辑的 svg 就好了

**Comment by wilsonwu** [comment]
> 这个插图的文字真多，能转成可编辑的 svg 就好了

如果原作者通过源文件生成 svg 就比较好办。

**Comment by windsonsea** [comment]
确实是，现在很尴尬

File: content/zh/blog/2023/egress-sni/index.md [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+无需配置每个域的 `VirtualService` 或其他路由详细信息。但要小心的是，
+由于域名在配置中的多个位置被列出。如果您使用 CI/CD 工具（例如 Kustomize），
+最好将域名列表提取到一个单一位置，使其被渲染到所需的配置资源中。
+
+就是这些！我希望这可以帮到您。如果您是之前基于 Nginx 解决方案的现有用户，
+现在可以在升级到 Istio 1.20 之前迁移到此方法，否则您当前的设置会被破坏。
+
+请开心的使用 SNI 路由！
+
+## References
```
Comment (hanxiaop) [inline content/zh/blog/2023/egress-sni/index.md:451]: > Need help:
> 
> ```
> Further documentation is available for these failures:
>  - MD022: https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md022---headers-should-be-surrounded-by-blank-lines
> ./zh/blog/2023/egress-sni/index.md:14:1.13 版的[官方 Istio 文档](https://istio.io/v1.13/zh/docs/tasks/traffic-management/egress/wildcard-egress-hosts/#wildcard-configuration-for-arbitrary-domains)中，
> ```
> 
> How to solve this lint problem.

Is it related to this extra line? Not sure though.

Comment (wilsonwu) [reply content/zh/blog/2023/egress-sni/index.md:451]: Sorry, the full message is:
```
Further documentation is available for these failures:
 - MD022: https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md022---headers-should-be-surrounded-by-blank-lines
./zh/blog/2023/egress-sni/index.md:14:1.13 版的[官方 Istio 文档](https://istio.io/v1.13/zh/docs/tasks/traffic-management/egress/wildcard-egress-hosts/#wildcard-configuration-for-arbitrary-domains)中，
**Ensure markdown content uses relative references to istio.io**
Running ["HtmlCheck", "ImageCheck", "OpenGraphCheck", "ScriptCheck"] on ["./public"] on *.html... 
```

File: content/zh/blog/2023/egress-sni/index.md [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+attribution: "Gergő Huszty (IBM); Translated by Wilson Wu (DaoCloud)"
+keywords: [traffic-management,gateway,mesh,mtls,egress,remote]
+---
+
+如果您使用 Istio 处理应用程序发起的流向网格外部目标的流量，您可能熟悉出口网关的概念。
+出口网关可用于监控流量并将其从网格内部应用程序转发到网格外部的位置。
+如果您的系统在受限环境中运行并且您想要控制从网格可以被访问的公共互联网内容，那么这是一个有用的功能。
+
+配置出口网关来处理任意通配符域的用例已包含在截止至
+1.13 版的[官方 Istio 文档](https://istio.io/v1.13/zh/docs/tasks/traffic-management/egress/wildcard-egress-hosts/#wildcard-configuration-for-arbitrary-domains)中，
```
[the remark this answers is not in the document]
Comment (wilsonwu) [reply content/zh/blog/2023/egress-sni/index.md]: Thanks for your help.

File: content/zh/blog/2023/egress-sni/index.md [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+title: "将出口流量路由到具有通配符的目标"
+description: "一种设置出口网关的通用方法，该网关可以动态地将流量路由到一组受限的目标远程主机（包括通配符域）。"
+publishdate: 2023-12-01
+attribution: "Gergő Huszty (IBM); Translated by Wilson Wu (DaoCloud)"
+keywords: [traffic-management,gateway,mesh,mtls,egress,remote]
+---
+
+如果您使用 Istio 处理应用程序发起的流向网格外部目标的流量，您可能熟悉出口网关的概念。
+出口网关可用于监控流量并将其从网格内部应用程序转发到网格外部的位置。
+如果您的系统在受限环境中运行并且您想要控制从网格可以被访问的公共互联网内容，那么这是一个有用的功能。
```
Comment (hanxiaop) [inline content/zh/blog/2023/egress-sni/index.md:11]:
```suggestion
如果您的系统在受限环境中运行并且您想控制从您的网格访问公共互联网的内容，那么这是一个有用的功能。
```

File: content/zh/blog/2023/egress-sni/index.md [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+
+配置出口网关来处理任意通配符域的用例已包含在截止至
```
Comment (hanxiaop) [inline content/zh/blog/2023/egress-sni/index.md:13]:
```suggestion
在[官方Istio文档](https://archive.istio.io/v1.13/zh/docs/tasks/traffic-management/egress/wildcard-egress-hosts/#wildcard-configuration-for-arbitrary-domains)中，配置出口网关以处理任意通配符域名的用例一直包含至1.13版本，
```

File: content/zh/blog/2023/egress-sni/index.md [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+1.13 版的[官方 Istio 文档](https://archive.istio.io/v1.13/zh/docs/tasks/traffic-management/egress/wildcard-egress-hosts/#wildcard-configuration-for-arbitrary-domains)中，
```
Comment (hanxiaop) [inline content/zh/blog/2023/egress-sni/index.md:14]:
```suggestion
```

Comment (hanxiaop) [reply content/zh/blog/2023/egress-sni/index.md:14]: or split lines maybe, not sure

File: content/zh/blog/2023/egress-sni/index.md [hunk 6]
```diff
@@ -0,0 +1,454 @@
...
+但随后已被删除，因为记录的解决方案未得到官方支持或推荐，
+并且在 Istio 的未来版本中可能会被破坏。尽管如此，
```
Comment (hanxiaop) [inline content/zh/blog/2023/egress-sni/index.md:16]:
```suggestion
并可能在未来的Istio版本中出现问题而被移除。尽管如此，
```

File: content/zh/blog/2023/egress-sni/index.md [hunk 7]
```diff
@@ -0,0 +1,454 @@
...
+keywords: [traffic-management,gateway,mesh,mtls,egress,remote]
+---
+
+如果您使用 Istio 处理应用程序发起的流向网格外部目标的流量，您可能熟悉出口网关的概念。
+出口网关可用于监控流量并将其从网格内部应用程序转发到网格外部的位置。
+如果您的系统在受限环境中运行并且您想要控制从网格可以被访问的公共互联网内容，那么这是一个有用的功能。
+
+配置出口网关来处理任意通配符域的用例已包含在截止至
+1.13 版的[官方 Istio 文档](https://archive.istio.io/v1.13/zh/docs/tasks/traffic-management/egress/wildcard-egress-hosts/#wildcard-configuration-for-arbitrary-domains)中，
+但随后已被删除，因为记录的解决方案未得到官方支持或推荐，
```
Comment (hanxiaop) [inline content/zh/blog/2023/egress-sni/index.md:15]:
```suggestion
但随后因为记录的解决方案未得到官方支持或推荐，
```

File: content/zh/blog/2023/egress-sni/index.md [hunk 8]
```diff
@@ -0,0 +1,454 @@
...
+并且在 Istio 的未来版本中可能会被破坏。尽管如此，
+旧的解决方案仍然可以在 1.20 之前的 Istio 版本中使用。
+然而，在 Istio 1.20 中放弃了支撑该方法工作所需的一些 Envoy 功能。
```
Comment (hanxiaop) [inline content/zh/blog/2023/egress-sni/index.md:18]:
```suggestion
然而，在 Istio 1.20 中放弃了一些该方法所需的 Envoy 的功能。
```

File: content/zh/blog/2023/egress-sni/index.md [hunk 9]
```diff
@@ -0,0 +1,454 @@
...
+
+本文试图描述我们如何解决这个问题，并通过使用 Istio 版本无关的组件和 Envoy
```
Comment (hanxiaop) [inline content/zh/blog/2023/egress-sni/index.md:20]:
```suggestion
本文试图描述我们如何解决这个问题，并通过使用与 Istio 版本独立的组件和 Envoy
```

File: content/zh/blog/2023/egress-sni/index.md [hunk 10]
```diff
@@ -0,0 +1,454 @@
...
+功能的类似方法来填补空白，但也不需要单独的 Nginx SNI 代理。
```
Comment (hanxiaop) [inline content/zh/blog/2023/egress-sni/index.md:21]:
```suggestion
功能的类似方法来填补空白，而无需单独的 Nginx SNI 代理。
```

File: content/zh/blog/2023/egress-sni/index.md [hunk 11]
```diff
@@ -0,0 +1,454 @@
...
+我们的方法允许旧解决方案的用户在其系统面临 Istio 1.20 中的重大变化之前无缝迁移配置。
+
+## 需要解决的问题 {#problem-to-solve}
+
+当前记录的出口网关用例依赖于这样一个事实：流量的目标（主机名）是在 `VirtualService` 中静态配置的，
```
Comment (hanxiaop) [inline content/zh/blog/2023/egress-sni/index.md:26]:
```suggestion
当前记录的出口网关用例依赖于流量的目标（主机名）是在 `VirtualService` 中静态配置的，
```

File: content/zh/blog/2023/egress-sni/index.md [hunk 12]
```diff
@@ -0,0 +1,454 @@
...
+并告知出口网关 Pod 中的 Envoy 在哪里进行 TCP 代理匹配的出站连接。
+您可以使用多个（甚至是通配符）DNS 名称来匹配路由条件，
+但您无法将流量路由到应用程序请求中指定的确切位置。例如，您可以匹配目标 `*.wikipedia.org` 的流量，
+但随后需要将流量转发到单个最终目标，例如 `en.wikipedia.org`。
+如果存在另一个服务，例如 `anyservice.wikipedia.org`，
+它不是由与 `en.wikipedia.org` 相同的服务器托管的，则到该主机的流量将会失败。
+这是因为，即使 HTTP 负载的 TLS 握手中的目标主机名包含 `anyservice.wikipedia.org`，
+但是 `en.wikipedia.org` 服务器也将无法响应该请求。
```
Comment (hanxiaop) [inline content/zh/blog/2023/egress-sni/index.md:34]:
```suggestion
`en.wikipedia.org` 服务器也无法响应该请求。
```

File: content/zh/blog/2023/egress-sni/index.md [hunk 13]
```diff
@@ -0,0 +1,454 @@
...
+
+此问题的高级解决方案是在每个新的网关连接中检查应用程序 TLS
```
Comment (hanxiaop) [inline content/zh/blog/2023/egress-sni/index.md:36]:
```suggestion
此问题的高级解决方案是在每个新的网关连接中检查应用程序 TLS
```

File: content/zh/blog/2023/egress-sni/index.md [hunk 14]
```diff
@@ -0,0 +1,454 @@
...
+握手（以明文形式发送，因此不需要 TLS 终止或其他中间人操作）
+中的原始服务器名称（SNI 扩展），并使用它作为离开网关的流量动态 TCP 代理目标。
```
Comment (hanxiaop) [inline content/zh/blog/2023/egress-sni/index.md:38]:
```suggestion
并将其用作动态 TCP 代理离开网关的流量的目标。
```

File: content/zh/blog/2023/egress-sni/index.md (hunk 4 above)
Comment (wilsonwu) [reply content/zh/blog/2023/egress-sni/index.md]: Applied.

File: content/zh/blog/2023/egress-sni/index.md (hunk 5 above)
Comment (wilsonwu) [reply content/zh/blog/2023/egress-sni/index.md]: Applied.

File: content/zh/blog/2023/egress-sni/index.md (hunk 6 above)
Comment (wilsonwu) [reply content/zh/blog/2023/egress-sni/index.md]: Applied.

File: content/zh/blog/2023/egress-sni/index.md [hunk 15]
```diff
@@ -0,0 +1,454 @@
...
+您可以使用多个（甚至是通配符）DNS 名称来匹配路由条件，
+但您无法将流量路由到应用程序请求中指定的确切位置。例如，您可以匹配目标 `*.wikipedia.org` 的流量，
+但随后需要将流量转发到单个最终目标，例如 `en.wikipedia.org`。
+如果存在另一个服务，例如 `anyservice.wikipedia.org`，
+它不是由与 `en.wikipedia.org` 相同的服务器托管的，则到该主机的流量将会失败。
+这是因为，即使 HTTP 负载的 TLS 握手中的目标主机名包含 `anyservice.wikipedia.org`，
+但是 `en.wikipedia.org` 服务器也将无法响应该请求。
+
+此问题的高级解决方案是在每个新的网关连接中检查应用程序 TLS
+握手（以明文形式发送，因此不需要 TLS 终止或其他中间人操作）
```
[the remark this answers is not in the document]
Comment (wilsonwu) [reply content/zh/blog/2023/egress-sni/index.md]: Applied.

File: content/zh/blog/2023/egress-sni/index.md [hunk 16]
```diff
@@ -0,0 +1,454 @@
...
+# 如果 SNI 目标主机名匹配，
+# VirtualService 将指示网格中的 Sidecar 将传出流量路由到出口网关服务
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: direct-wildcard-through-egress-gateway
+  namespace: istio-system
+spec:
+  hosts:
+    - "*.wikipedia.org"
```
Comment (hanxiaop) [inline content/zh/blog/2023/egress-sni/index.md:189]: indentation 好像不太统一

Comment (wilsonwu) [reply content/zh/blog/2023/egress-sni/index.md:189]: Try to make all yaml as standard, but seems it is not a small work, need to do more verification after change format.

**Pull request merged**